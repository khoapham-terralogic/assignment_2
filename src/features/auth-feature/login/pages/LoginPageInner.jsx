import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import { ClipSpinner } from '../../../../components'
import LoginForm from '../components/LoginForm'

const LoginPageInner = ({ isLoading, loginUser, isAuth }) => {
    const history = useHistory()

    useEffect(() => {
        if (isAuth) history.push("/user")
    })
    return (
        <div className="login">
            {isLoading ? <div className="loadingBackground"><ClipSpinner size={45} color="#fff" /></div> : null}
            <div className="login-header">login your account</div>
            <LoginForm loginUser={loginUser} isLoading={isLoading} />
        </div>
    );
}

LoginPageInner.propTypes = {
    isLoading: PropTypes.bool.isRequired,
    isAuth: PropTypes.bool.isRequired,
    loginUser: PropTypes.func.isRequired,
}

export default LoginPageInner
