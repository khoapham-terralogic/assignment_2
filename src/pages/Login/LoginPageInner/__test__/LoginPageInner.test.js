import React from 'react'
import { shallow } from 'enzyme'
import LoginPageInner from '..';

describe('Test <LoginPageInner />', () => {
    const props = {
        isAuth: false,
        isLoading: false,
        loginUser: jest.fn()
    }
    const wrapper = shallow(<LoginPageInner {...props} />)
    it('should render', () => {
        expect(wrapper).toMatchSnapshot()
    });
    it('should render with required props', () => {
        expect(wrapper.prop('isAuth')).toBeFalsy()
        expect(wrapper.prop('isLoading')).toBeFalsy()
        expect(wrapper).toMatchSnapshot()
    });
    it('should render loading if isLoading is true', () => {
        const wrap = shallow(<LoginPageInner isLoading={true} isAuth={false} loginUser={jest.fn()} />)
        wrapper.update()
        expect(wrap.find('.loadingBackground')).toHaveLength(1)
    });
    it('should push to /user if isAuth is true', () => {
        let useEffect
        const mockUseEffect = () => {
            useEffect.mockImplementationOnce(f => f());
        };
        beforeEach(() => {
            const wrapper = shallow(<LoginPageInner isLoading={false} isAuth={true} loginUser={jest.fn()} />);
            useEffect = jest.spyOn(React, "useEffect").mockImplementation(f => f());
            mockUseEffect();
            mockUseEffect();
            expect(wrapper.find(LoginPageInner)).toHaveLength(0)
        });
    })
})
