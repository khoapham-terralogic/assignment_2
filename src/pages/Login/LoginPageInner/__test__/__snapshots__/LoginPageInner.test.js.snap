// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <LoginPageInner /> should render 1`] = `
<div
  className="login"
>
  <div
    className="login-header"
  >
    login your account
  </div>
  <WithFormik(LoginInnerForm)
    isLoading={false}
    loginUser={[MockFunction]}
  />
</div>
`;

exports[`Test <LoginPageInner /> should render with required props 1`] = `
<div
  className="login"
>
  <div
    className="login-header"
  >
    login your account
  </div>
  <WithFormik(LoginInnerForm)
    isLoading={false}
    loginUser={[MockFunction]}
  />
</div>
`;
